lib/middleware.js

```diff
@@ -2,17 +2,18 @@
 
 // Creates new middleware using provided options
 function create (rules) {
-  return function redirectRoute (req, res, next) {
+  return async function redirectRoute (req, res, next) {
     const decodedBaseUrl = decodeURI(req.url)
     const foundRule = rules.find(o => o.from.test(decodedBaseUrl))
 
     if (!foundRule) {
       return next()
     }
-    // Expect rule 'to' to be 1)regex or 2)string or 3)synchronous function taking from to compute to
+    // Expect rule 'to' to be 1)regex or 2)string or
+    // 3)function taking from & req (when from is regex, req might be more interesting)
```

Comment:
refer to the new test case

Reply:
Giving in the `req` as 2nd argument is smart! Like that ☺️